src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHelper.cs

```diff
@@ -35,6 +35,20 @@ internal static class DiagnosticsHelper
             _ => httpVersion.ToString()
         };
 
+        // Picks the value of the 'server.address' tag following rules specified in
+        // https://github.com/open-telemetry/semantic-conventions/blob/728e5d1/docs/http/http-spans.md#http-client-span
+        // When there is no proxy, we need to prioritize the contents of the Host header.
+        public static string GetServerAddress(HttpRequestMessage request, IWebProxy? proxy)
+        {
+            Debug.Assert(request.RequestUri is not null);
+            if ((proxy is null || proxy.IsBypassed(request.RequestUri)) && request.HasHeaders && request.Headers.Host is string hostHeader)
```

Comment:
`IWebProxy.GetProxy` can return null even if `IsBypassed` returns false.

I take it this is just a some/best-effort guess if we'll be using a proxy since we don't have a great wey to flow the real answer down to the pool manager?